app/src/main/java/org/feature/fox/coffee_counter/ui/user/UserListViewModel.kt

```diff
@@ -168,14 +169,25 @@ class UserListViewModel @Inject constructor(
             return
         }
 
-        response.data.forEach { user ->
-            val idResponse = userRepository.getUserById(user.id)
+        if(isAdminState.value) {
+            response.data.forEach { user ->
+                val idResponse = userRepository.getUserById(user.id)
 
-            if (idResponse.data == null) {
-                return@forEach
+                if (idResponse.data == null) {
+                    return@forEach
+                }
+
+                userList.add(idResponse.data)
+            }
+        }else{
+            response.data.forEach { user ->
+                userList.add(UserIdResponse(
+                    id = user.id,
+                    name = user.name,
+                    balance = 0.0,
+                ))
```

Comment:
In both cases you are interating over the same list, you can move the if-else inside one iteration and get rid of the other.